build.assets/kubectl-version/main.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+	"runtime/debug"
+	"strings"
+
+	_ "k8s.io/kubectl"
+)
+
+func main() {
+	if version, ok := getKubectlModVersion(); ok {
+		fmt.Println(version)
+	}
+}
+
+func getKubectlModVersion() (string, bool) {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return "", false
+	}
+	for _, dep := range info.Deps {
+		if dep.Path == "k8s.io/kubectl" {
+			return strings.Replace(dep.Version, "v0.", "v1.", 1), true
+		}
+	}
+	return "", false
+}
```

Comment:
I'm probably missing some context here, but is there a reason why this can't be evaluated by tsh at run time, rather than adding this build-time logic?